client/lib/sites-list/sites-observer.jsx

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import omit from 'lodash/omit';
+
+/**
+ * Internal dependencies
+ */
+import Debug from 'debug';
+const debug = Debug( 'calypso:site-observe-hoc' );
+
+export const sitesObserver = ( WrappedComponent ) => {
+	class SitesObserverComponent extends Component {
+
+		constructor( props, context ) {
+			super( props, context );
+
+			this.update = this.update.bind( this );
+
+			this.state = {};
+		}
+
+		componentDidMount() {
+			this.props.sites.on( 'change', this.update );
+			this.cacheSites();
+		}
+
+		componentWillUnmount() {
+			this.props.sites.off( 'change', this.update );
+		}
+
+		update() {
+			debug( 'Re-rendering ' + this.constructor.displayName + ' component.' );
+			this.cacheSites();
+		}
+
+		cacheSites() {
+			if ( this.props.sites ) {
+				this.setState( { sites: Object.create( this.props.sites ) } );
```

Comment:
I've checked at it looks to me like it is all fine.

1. Right after [SitesList is instantiated](https://github.com/Automattic/wp-calypso/blob/master/client/lib/sites-list/index.js#L14) it is added to PollerPool which in turn [create a new Poller](https://github.com/Automattic/wp-calypso/blob/master/client/lib/data-poller/index.js#L7) [which register to events](https://github.com/Automattic/wp-calypso/blob/master/client/lib/data-poller/poller.js#L33) on the newly created SitesList. In that `on` implementation the `SitesList` instance will get `undefined` value when accessing the prototype `_events` and will create a new property `_events` on the `SitesList` singleton instance. When we'll use `Object.create` to create a new instance that has its prototype point to that singleton `SitesList` instance any call to `on` and `off` will now always get the `SitesList` value of `_events` when accessing `this._events' where `this` is the new instance with `SitesList` as the prototype so any event addition or removal will be correctly accounted for on the `SitesList` singleton.

2. Specifically in `MySitesSidebar` as far as I checked all the sub-components rendered (directly or indirectly) are using `sites` prop without registering to events.